sklearn/feature_selection/from_model.py

```diff
@@ -199,9 +215,33 @@ def _get_support_mask(self):
                 'Either fit the model before transform or set "prefit=True"'
                 ' while passing the fitted estimator to the constructor.')
         scores = _get_feature_importances(estimator)
-        self.threshold_ = _calculate_threshold(estimator, scores,
-                                               self.threshold)
-        return scores >= self.threshold_
+        mask = np.zeros(scores.shape, dtype=bool)
+        if self.threshold is not None and self.max_n_features is not None:
+            self.threshold_ = _calculate_threshold(estimator, scores,
+                                                   self.threshold)
+            mask = scores >= self.threshold_
+            if self.max_n_features == 'all':
+                pass
+            elif self.max_n_features == 0:
+                mask[:] = False
+            else:
+                scores = np.ma.masked_array(scores, mask=1-mask)
+                max_n_indices = scores.argsort()[:scores.count()][:self.
```

Comment:
The argsort is not stable. I presume it is deterministic, but you could not explain why one feature or another is chosen when there is a tie.